test/suites/smoke-test-common-solo/test-staking-requests.ts

```diff
@@ -0,0 +1,153 @@
+                        }
+                    }
+                }
+
+                // We record the session too as we need to compare it against the session at which the request was injected
+                return {
+                    blockNum: blockNum,
+                    extrinsics: signedBlock.block.extrinsics,
+                    events: await apiAt.query.system.events(),
+                    session: await apiAt.query.session.currentIndex(),
+                    pendingOperations: operatorPendingOps,
+                };
+            };
+            const limiter = new Bottleneck({ maxConcurrent: 5, minTime: 100 });
+            blockData = await Promise.all(blockNumArray.map((num) => limiter.schedule(() => getBlockData(num))));
+        }, timeout);
+
+        it({
+            id: "C01",
+            title: "If someone executes a pending request, it should have been generated at least 2 sessions ago",
+            test: async () => {
+                const filteredEvents = blockData
+                    .map(({ blockNum, events, session, pendingOperations }) => {
+                        const matchedEvents = events
+                            .filter(({ event }) => api.events.system.ExtrinsicSuccess.is(event))
+                            .filter(({ event }) => {
+                                const info = event.data[0] as DispatchInfo;
+                                return info.class.isNormal && info.paysFee.isYes;
+                            });
+                        return { blockNum, matchedEvents, session, pendingOperations };
+                    })
+                    .filter(({ matchedEvents }) => matchedEvents.length > 0);
+
+                // This function evaluates whether there exists at least one pending op that has been executed
+                const isValidPolledStakingOp = async (blockNum: number, index: number, session, pendingOperations) => {
+                    const extrinsic = blockData.find((a) => a.blockNum === blockNum)!.extrinsics[index];
+                    const isValidExtrinsic =
+                        extrinsic.method.section.toString() === "pooledStaking" &&
+                        extrinsic.method.method.toString() === "executePendingOperations";
+
+                    if (isValidExtrinsic) {
+                        const injectedOps = extrinsic.args[0].toHuman();
+
+                        for (let i = 0; i < injectedOps.length; i++) {
+                            const operation = injectedOps[i];
+                            const pendingBefore = pendingOperations[operation.delegator].map(([_, op]) => op);
+
+                            const time = Number.parseInt(
+                                operation.operation[Object.keys(operation.operation)[0]].at.replace(",", "")
+                            );
+
+                            const filtered = pendingBefore.filter(
+                                (op) => JSON.stringify(op) === JSON.stringify(operation.operation)
```

Comment:
Can we guarantee order here? Otherwise, it won't be equal.

Reply:
it "should" only be single element no vector. so I guess it works for now